File: CS/IReportDesignerAPIService.cs
using DevExpress.Mvvm.UI;
using DevExpress.Mvvm.UI.Interactivity;
using DevExpress.Xpf.Reports.UserDesigner;
using DevExpress.XtraReports.UI;
using System;
using System.Collections.Generic;
using System.IO;

namespace ManipulateReportDesignerDocuments_MVVM {
    public interface IReportDesignerAPIService {
        IEnumerable<ReportDesignerDocument> Documents { get; }
        ReportDesignerDocument ActiveDocument { get; }
        ReportDesignerDocument NewReport(XtraReport report = null);
        ReportDesignerDocument Open();
        ReportDesignerDocument Open(Stream stream);
    }

    public class ReportDesignerAPIService : ServiceBase, IReportDesignerAPIService {
        ReportDesigner Designer => (ReportDesigner)AssociatedObject;

        public IEnumerable<ReportDesignerDocument> Documents => Designer.Documents;

        public ReportDesignerDocument ActiveDocument => Designer.ActiveDocument;

        public ReportDesignerDocument NewReport(XtraReport report = null) {
            Func<XtraReport> reportFactory = null;
            if(report != null)
                reportFactory = () => report;
            return Designer.NewDocument(reportFactory);
        }

        public ReportDesignerDocument Open() {
            return Designer.OpenDocument();
        }

        public ReportDesignerDocument Open(Stream stream) {
            return Designer.OpenDocument(stream);
        }
    }
}

File: VB/IReportDesignerAPIService.vb
Imports DevExpress.Mvvm.UI
Imports DevExpress.Mvvm.UI.Interactivity
Imports DevExpress.Xpf.Reports.UserDesigner
Imports DevExpress.XtraReports.UI
Imports System
Imports System.Collections.Generic
Imports System.IO

Namespace ManipulateReportDesignerDocuments_MVVM
	Public Interface IReportDesignerAPIService
		ReadOnly Property Documents() As IEnumerable(Of ReportDesignerDocument)
		ReadOnly Property ActiveDocument() As ReportDesignerDocument
		Function NewReport(Optional ByVal report As XtraReport = Nothing) As ReportDesignerDocument
		Function Open() As ReportDesignerDocument
		Function Open(ByVal stream As Stream) As ReportDesignerDocument
	End Interface

	Public Class ReportDesignerAPIService
		Inherits ServiceBase
		Implements IReportDesignerAPIService

		Private ReadOnly Property Designer() As ReportDesigner
			Get
				Return CType(AssociatedObject, ReportDesigner)
			End Get
		End Property

		Public ReadOnly Property Documents() As IEnumerable(Of ReportDesignerDocument) Implements IReportDesignerAPIService.Documents
			Get
				Return Designer.Documents
			End Get
		End Property

		Public ReadOnly Property ActiveDocument() As ReportDesignerDocument Implements IReportDesignerAPIService.ActiveDocument
			Get
				Return Designer.ActiveDocument
			End Get
		End Property

		Public Function NewReport(Optional ByVal report As XtraReport = Nothing) As ReportDesignerDocument Implements IReportDesignerAPIService.NewReport
			Dim reportFactory As Func(Of XtraReport) = Nothing
			If report IsNot Nothing Then
				reportFactory = Function() report
			End If
			Return Designer.NewDocument(reportFactory)
		End Function

		Public Function Open() As ReportDesignerDocument Implements IReportDesignerAPIService.Open
			Return Designer.OpenDocument()
		End Function

		Public Function Open(ByVal stream As Stream) As ReportDesignerDocument Implements IReportDesignerAPIService.Open
			Return Designer.OpenDocument(stream)
		End Function
	End Class
End Namespace
